library/alloc/src/vec/drain.rs

```diff
@@ -112,7 +112,7 @@ impl<'a, T, A: Allocator> Drain<'a, T, A> {
             let unyielded_ptr = this.iter.as_slice().as_ptr();
 
             // ZSTs have no identity, so we don't need to move them around.
-            let needs_move = mem::size_of::<T>() != 0;
+            let needs_move = !T::IS_ZST;
 
             if needs_move {
```

Comment:
Minor style thing: As far as I can see this is the only use of the variable and it already has a comment so it could just be moved into the if condition.

Reply:
[Updated](https://github.com/rust-lang/rust/compare/048c135ecebe36409c202f55d4542668ba44130f..56613f8c382351898f9267bac8e8324f118eb79d) as requested.